pkg/beacon/analyzer.go

```diff
@@ -65,28 +65,41 @@ func (a *analyzer) start() {
 	go func() {
 		for res := range a.analysisChannel {
 
-			//store the diff slice length since we use it a lot
+			//store the diffFull slice length since we use it a lot
 			//for timestamps this is one less then the data slice length
 			//since we are calculating the times in between readings
 			tsLength := len(res.TsList) - 1
 			dsLength := len(res.OrigBytesList)
 
 			//find the delta times between the timestamps and sort
-			diff := make([]int64, tsLength)
+			//we are excluding delta zero for scoring calculations
+			//but using a separate array that includes it for making
+			//the user/graph reference variables returned by createCountMap
+			diffFull := make([]int64, tsLength)
+			var diff []int64
 			for i := 0; i < tsLength; i++ {
-				diff[i] = res.TsList[i+1] - res.TsList[i]
+				interval := res.TsList[i+1] - res.TsList[i]
+				diffFull[i] = interval
+				if interval > 0 {
+					diff = append(diff, interval)
+				}
+
 			}
 			sort.Sort(util.SortableInt64(diff))
+			sort.Sort(util.SortableInt64(diffFull))
```

Comment:
Performance wise, it would be faster to generate `diffFull`, sort it, and then loop through `diffFull` appending the non-zero entries to `diff`. This would remove the need to sort `diff` on its own.

Reply:
I've pushed up a fix for this. I am currently testing to ensure that no changes were made to the resulting scores.

Reply:
I have verified that none of the scores changed with this fix.